src/hrs.ml

```diff
@@ -42,30 +42,35 @@ let print_term : bool -> term pp = fun lhs ->
         out "pi(%a,\\v_%s.%a)" pp a (Bindlib.name_of x) pp b
   in pp
 
-(** [print_rule oc s r] outputs the rule declaration corresponding [r] (on the
-    symbol [s]), to the output channel [oc]. *)
-let print_rule : Format.formatter -> sym -> rule -> unit = fun oc s r ->
-  (* Print the pattern variable declarations. *)
-  let get_patt_names : term list -> string list = fun ts ->
+(** [print_rule oc lhs rhs] outputs the rule declaration [lhs]->[rhs]
+    to the output channel [oc] *)
+let print_rule : Format.formatter -> Terms.term -> Terms.term -> unit =
+  fun oc lhs rhs ->
+  (* gets pattern and binded variable names *)
+  let get_var_names : term -> string list = fun ts ->
     let names = Pervasives.ref [] in
     let fn t =
       match t with
-      | Patt(_,n,_) -> Pervasives.(names := n :: !names)
+      | Patt(_,n,_) -> Pervasives.(names := ("&" ^ n) :: !names)
+      | Abst(_,b) ->
+        let (x, _) = Bindlib.unbind b in
+        let n = Bindlib.name_of x in
+        Pervasives.(names := ("v_" ^ n) :: !names)
       | _           -> ()
     in
-    List.iter (Basics.iter fn) ts;
+    Basics.iter fn ts;
     List.sort_uniq String.compare Pervasives.(!names)
   in
-  let names = get_patt_names r.lhs in
+  let names_lhs = get_var_names lhs in
+  let names_rhs = get_var_names rhs in
+  let names = List.sort_uniq String.compare (names_lhs @ names_rhs) in
```

Comment:
remove call to List.sort_uniq by making get_var_names return an Extra.StrSet.t

Reply:
two occurences of ``List.sort_uniq`` have been replaced in b2da27c